### PR TITLE
`@0x/contracts-erc20-bridge-sampler`: LiquidityProviderRegistry

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Updates the way the ERC20BridgeSampler samples from generic liquidity providers to route through a registry. This allows the owner of the registry to swap out the underlying liquidity providers without interruption for 0x API

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol [hunk 1]
```diff
@@ -451,6 +452,12 @@ contract ERC20BridgeSampler is
         view
         returns (uint256[] memory makerTokenAmounts)
     {
+        address providerAddress = getLiquidityProviderFromRegistry(
```
Comment (PirosB3) [inline contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol]: This method [may revert](https://github.com/0xProject/plp/blob/development/contracts/src/PLPRegistry.sol#L75) if a liquidity pool does not exist. How do we handle reverts?

File: contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol [hunk 2]
```diff
@@ -512,6 +525,26 @@ contract ERC20BridgeSampler is
...
+    function getLiquidityProviderFromRegistry(
+        address registryAddress,
+        address takerToken,
+        address makerToken
+    )
+        public
+        view
+        returns (address)
+    {
+        return ILiquidityProviderRegistry(registryAddress).getLiquidityProviderForMarket(
```
Comment (PirosB3) [inline contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol]: This method [may revert](https://github.com/0xProject/plp/blob/development/contracts/src/PLPRegistry.sol#L75) if a liquidity pool does not exist. How do we handle reverts?

File: contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol [hunk 3]
```diff
@@ -451,6 +452,12 @@ contract ERC20BridgeSampler is
         view
         returns (uint256[] memory makerTokenAmounts)
     {
+        address providerAddress = getLiquidityProviderFromRegistry(
+            registryAddress,
+            takerToken,
+            makerToken
+        );
+
```
Comment (PirosB3) [inline contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol]: if a liquidity pool is not found, we should return an array of 0s

File: contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol [hunk 4]
```diff
@@ -435,14 +436,14 @@ contract ERC20BridgeSampler is
...
 
     /// @dev Sample sell quotes from an arbitrary on-chain liquidity provider.
-    /// @param providerAddress Address of the liquidity provider contract.
+    /// @param registryAddress Address of the liquidity provider registry contract.
     /// @param takerToken Address of the taker token (what to sell).
     /// @param makerToken Address of the maker token (what to buy).
     /// @param takerTokenAmounts Taker token sell amount for each sample.
     /// @return makerTokenAmounts Maker amounts bought at each taker token
     ///         amount.
     function sampleSellsFromLiquidityProvider(
```
Comment (PirosB3) [inline contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol]: As discussed, this should be `sample(Sells|Buys)FromLiquidityProviderRegistry()`

File: contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol [hunk 5]
```diff
@@ -512,6 +541,32 @@ contract ERC20BridgeSampler is
...
+    /// @param makerToken Maker asset managed by liquidity provider.
+    /// @return providerAddress Address of the liquidity provider.
+    function getLiquidityProviderFromRegistry(
+        address registryAddress,
+        address takerToken,
+        address makerToken
+    )
+        public
+        view
+        returns (address providerAddress)
```
Comment (PirosB3) [inline contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol]: by default, is `providerAddress` set to `address(0)`?

Comment (mzhu25) [reply contracts/erc20-bridge-sampler/contracts/src/ERC20BridgeSampler.sol]: yep

**Pull request merged**